infrastructure/zk/src/database.ts

```diff
@@ -1,101 +1,233 @@
 import { Command } from 'commander';
 import * as utils from './utils';
 
-export async function reset() {
+export async function reset(opts: any) {
     await utils.confirmAction();
-    await wait();
-    await drop();
-    await setup();
+    await wait(opts);
+    await drop(opts);
+    await setup(opts);
 }
 
-export async function resetTest() {
-    process.env.DATABASE_URL = process.env.TEST_DATABASE_URL;
-    await utils.confirmAction();
+enum DalPath {
+    CoreDal = 'core/lib/dal',
+    ProverDal = 'prover/prover_dal'
+}
+
+function getDals(opts: any): Map<DalPath, string> {
+    let dals = new Map<DalPath, string>();
+    if (!opts.prover && !opts.server) {
+        dals.set(DalPath.CoreDal, process.env.DATABASE_URL!);
+        dals.set(DalPath.ProverDal, process.env.DATABASE_PROVER_URL!);
+    }
+    if (opts.prover) {
+        dals.set(DalPath.ProverDal, process.env.DATABASE_PROVER_URL!);
+    }
+    if (opts.server) {
+        dals.set(DalPath.CoreDal, process.env.DATABASE_URL!);
+    }
+    return dals;
+}
+
+function getTestDals(opts: any): Map<DalPath, string> {
+    let dals = new Map<DalPath, string>();
+    if (!opts.prover && !opts.server) {
+        dals.set(DalPath.CoreDal, process.env.TEST_DATABASE_URL!);
+        dals.set(DalPath.ProverDal, process.env.TEST_DATABASE_PROVER_URL!);
+    }
+    if (opts.prover) {
+        dals.set(DalPath.ProverDal, process.env.TEST_DATABASE_PROVER_URL!);
+    }
+    if (opts.server) {
+        dals.set(DalPath.CoreDal, process.env.TEST_DATABASE_URL!);
+    }
+    return dals;
+}
+
+async function resetTestDal(dalPath: DalPath, dbUrl: string) {
     console.log('recreating postgres container for unit tests');
     await utils.spawn('docker compose -f docker-compose-unit-tests.yml down');
     await utils.spawn('docker compose -f docker-compose-unit-tests.yml up -d');
     await wait(100);
     console.log('setting up a database template');
-    await setup();
+    await setupForDal(dalPath, dbUrl);
     console.log('disallowing connections to the template');
     await utils.spawn(
-        `psql "${process.env.DATABASE_URL}" -c "update pg_database set datallowconn = false where datname = current_database()"`
+        `psql "${dbUrl}" -c "update pg_database set datallowconn = false where datname = current_database()"`
     );
 }
 
-export async function drop() {
+export async function resetTest(opts: any) {
```

Comment:
Btw, why `any`? We can declare an interface for that to be typesafe.

Though if we will consider an approach with a public enum for the dal type, I think we don't need it at all. We can either take an array of `DBType` or even let the caller deal with it (if they have to, they can just call the method twice, once for `Server` and once for `Prover`.